neps/search_spaces/yaml_search_space_utils.py

```diff
@@ -0,0 +1,362 @@
+import re
+
+
+def convert_scientific_notation(value, show_usage_flag=False):
+    """Check if the value is a string that matches scientific e notation and convert it
+    to float. (specially numbers like 3.3e-5 with a float value in front, which yaml
+    can not interpret directly as float)."""
+
+    e_notation_pattern = r"^-?\d+(\.\d+)?[eE]-?\d+$"
+
+    flag = False  # Flag if e notation was detected
+
+    if isinstance(value, str):
+        # Remove all whitespace from the string
+        value_no_space = value.replace(" ", "")
+
+        # check for e notation
+        if re.match(e_notation_pattern, value_no_space):
+            flag = True
+
+    if show_usage_flag is True:
+        return float(value), flag
+    else:
+        return float(value)
+
+
+class SearchSpaceFromYamlFileError(Exception):
+    """
+    Exception raised for errors occurring during the initialization of the search space
+    from a YAML file.
+
+    Attributes:
+        exception_type (str): The type of the original exception.
+        message (str): A detailed message that includes the type of the original exception
+                       and the error description.
+
+    Args:
+        exception (Exception): The original exception that was raised during the
+                                initialization of the search space from the YAML file.
+
+    Example Usage:
+        try:
+            # Code to initialize search space from YAML file
+        except (KeyError, TypeError, ValueError) as e:
+            raise SearchSpaceFromYamlFileError(e)
+    """
+
+    def __init__(self, exception):
+        self.exception_type = type(exception).__name__
+        self.message = (
+            f"Error occurred during initialization of search space from "
+            f"YAML file.\n {self.exception_type}: {exception}"
+        )
+        super().__init__(self.message)
+
+
+def deduce_and_validate_param_type(name, details):
+    """
+    Deduces the parameter type from details and validates them.
+
+    Args:
+        name (str): The name of the parameter.
+        details (dict): A dictionary containing parameter specifications.
```

Comment:
Might be a bit nitpicking here but since the argument types are available in the comments would it also make sense to have them in the function arguments themselves ?
ex:
```python
def deduce_and_validate_param_type(name: str, details: dict):
```
Or: 
```python
from typing import Dict

def deduce_and_validate_param_type(name: str, details: Dict[str, int]): # Any types required
```

Reply:
Agree, added it + some missing details in some parameter validation functions[here](https://github.com/automl/neps/pull/46/commits/91d8a457db76cee1970fbabd6941325c656b758e)